src/main/app/pay/payClient.ts

```diff
@@ -31,6 +31,88 @@ export class PayClient {
    * @param returnURL - the url the user should be redirected to
    * @returns response with payment status and link to card payment page
    */
+  create (user: User, caseReference: string, externalId: string, fees: Fee[], returnURL: string): Promise<Payment>
+
+  /**
+   * Retrieves a payment from Reform Payment Hub
+   *
+   * @param user - user who makes a call
+   * @param paymentReference - payment reference number within Reform Payment Hub, generated when payment was created
+   * @returns response all payment details including most recent payment status or undefined when reference does not exist
+   */
+  retrieve (user: User, paymentReference: string): Promise<PaymentRetrieveResponse | undefined>
+}
+
+const delay = ms => new Promise(_ => setTimeout(_, ms))
+
+export class MockPayClient implements PayClient {
+  constructor (public serviceAuthToken: ServiceAuthToken, public requestUrl: string) {}
+
+  async create (user: User, caseReference: string, externalId: string, fees: Fee[], returnURL: string): Promise<Payment> {
+    /*
+    Calculated from:
+     dependencies
+      | where name == "POST /card-payments"
+      | summarize avg(duration)
+     */
+    const payCreateDelayInMs = 694
+    await delay(payCreateDelayInMs)
+
+    const reference = `RC-${this.referencePart()}-${this.referencePart()}-${this.referencePart()}-${this.referencePart()}`
+    return Promise.resolve({
+      reference: reference,
+      date_created: MomentFactory.currentDateTime().toISOString(),
+      status: 'Initiated',
+      _links: {
+        next_url: {
+          href: `${this.requestUrl}/${externalId}/receiver`,
+          method: 'GET'
+        }
+      }
+    })
+  }
+
+  private referencePart (): number {
+    return Math.floor(1000 + Math.random() * 9000)
+  }
+
+  async retrieve (user: User, paymentReference: string): Promise<PaymentRetrieveResponse | undefined> {
+    const payRetrieveDelayInMs = 681
+    /*
+    Calculated from:
+     dependencies
+      | where name contains "GET /card-payments"
+      | summarize avg(duration)
+     */
+    await delay(payRetrieveDelayInMs)
+
+    return Promise.resolve(
+      {
+        amount: 185,
+        description: description,
+        reference: paymentReference,
+        currency: 'GBP',
+        caseReference: uuid(),
+        channel: 'online',
+        method: 'card',
+        externalProvider: 'gov pay',
+        status: 'Success',
+        externalReference: 'a-gov-pay-reference',
+        siteId: siteId,
+        serviceName: serviceName,
+        fees: [
+          { calculated_amount: 185, code: 'X0029', version: '1' }
+        ]
+      })
+  }
+
+}
+
+export class GovPayClient implements PayClient {
+  constructor (public serviceAuthToken: ServiceAuthToken) {
```

Comment:
An improvement to existing code.

A `serviceAuthToken` does not have to be public if we do `this.serviceAuthToken = serviceAuthToken`.

Could be either:

```
constructor (private serviceAuthToken: ServiceAuthToken) {}
```

or

```
constructor (serviceAuthToken: ServiceAuthToken) {
  this.serviceAuthToken = serviceAuthToken
}
```